spec.bs

```diff
@@ -575,17 +574,32 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
     [=parallel queue/enqueue steps|enqueue the following steps=] to |queue|:
     1. Let |compWinner| be the result of [=generate and score bids=] with |component|,
       |auctionConfig|, and |global|.
-    1. If |compWinner| is not failure or null, [=score and rank a bid=] with |auctionConfig|,
-      |compWinner|, |leadingBidInfo|, |desisionLogicScript|, and true.
+    1. If |compWinner| is not null:
+      1. Run [=score and rank a bid=] with |auctionConfig|, |compWinner|, |leadingBidInfo|,
+        |desisionLogicScript|, and true.
     1. Decrement |pendingComponentAuctions| by 1.
   1. Wait until |pendingComponentAuctions| is 0.
+  1. If |leadingBidInfo|'s [=leading bid info/leading bid=] is null, return null.
+  1. Let |winningComponentConfig| be |leadingBidInfo|'s [=leading bid info/auction config=].
+  1. Set |leadingBidInfo|'s [=leading bid info/auction config=] to |auctionConfig|.
+  1. Let « |topLevelSellerSignals|, topLevelReportResultBrowserSignals » be the result of running
+    [=report result=] with |leadingBidInfo|.
+  1. Set |leadingBidInfo|'s [=leading bid info/auction config=] to |winningComponentConfig|.

```

Comment:
Why do we set the leading auction config to |auctionConfig| on line 584, but then set it back to the original (winningComponentConfig|)? Is the former the top-level auction config, whereas |winningComponentConfig| belongs to a component auction?

Reply:
Yes, exactly, the first call to reportResult() is for the top-level auction, and the second is for the winning component auction.

Reply:
Do we use the top-level config for reporting?